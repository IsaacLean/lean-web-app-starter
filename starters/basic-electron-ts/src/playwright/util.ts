import {
  ElectronApplication,
  Page,
  _electron as electron,
} from "@playwright/test";

/**
 * Display console API method calls from the Electron application in the Playwright console.
 * @param electronApp Electron application representation
 */
export const enableConsole = (electronApp: ElectronApplication) => {
  electronApp.on("console", (data) => console.log(data));
};

/**
 * Get the operating system that the browser is running in.
 * @param page Electron Page instance
 * @returns A promise that will resolve to the Browser's operating system type, or null if
 *     it could not be determined
 */
export const getBrowserOs = async (page: Page) => {
  const userAgent = await page.evaluate(async () => window.navigator.userAgent);

  if (userAgent.includes("Windows")) {
    return "win";
  } else if (userAgent.includes("Mac")) {
    return "mac";
  } else if (userAgent.includes("X11")) {
    return "unix";
  } else if (userAgent.includes("Linux")) {
    return "linux";
  }

  return null;
};

/**
 * Launch the Electron application with either the development or production build.
 * By default the PLAYWRIGHT_BUILD_TYPE environment variable will determine which build will be used,
 * but you can override this through some of the available parameters.
 * @param arg Parameter object
 * @param arg.buildPath Manually set the build path. Overrides buildType parameter and PLAYWRIGHT_BUILD_TYPE environment variable.
 * @param arg.buildType Determines if the development or the production build is used for testing. Overrides PLAYWRIGHT_BUILD_TYPE environment variable.
 * @param arg.launchOptions Options to pass to Electron.launch method
 * @returns A promise that will resolve to an Electron application representation
 */
export const launchElectron = ({
  buildPath,
  buildType,
  launchOptions,
}: {
  buildPath?: string;
  buildType?: "development" | "production";
  launchOptions?: Parameters<typeof electron.launch>[0];
} = {}) => {
  let arg = ".";
  if (buildPath) {
    arg = buildPath;
  } else if (
    buildType === "production" ||
    process.env.PLAYWRIGHT_BUILD_TYPE === "production"
  ) {
    arg = "./build/production/main/main.js";
  }

  return electron.launch({
    args: [arg],
    ...launchOptions,
  });
};
