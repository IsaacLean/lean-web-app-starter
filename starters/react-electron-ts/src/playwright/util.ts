import {
    ElectronApplication,
    Page,
    _electron as electron,
} from '@playwright/test'

/**
 * Display console API method calls from the Electron application in the Playwright console.
 * @param electronApp Electron application representation
 */
export const enableConsole = (electronApp: ElectronApplication) => {
    electronApp.on('console', (data) => console.log(data))
}

/**
 * Get the operating system that the browser is running in.
 * @param page Electron Page instance
 * @returns A promise that will resolve to the Browser's operating system type, or null if
 *     it could not be determined
 */
export const getBrowserOs = async (page: Page) => {
    const userAgent = await page.evaluate(
        async () => window.navigator.userAgent,
    )

    if (userAgent.includes('Windows')) {
        return 'win'
    } else if (userAgent.includes('Mac')) {
        return 'mac'
    } else if (userAgent.includes('X11')) {
        return 'unix'
    } else if (userAgent.includes('Linux')) {
        return 'linux'
    }

    return null
}

/**
 * Launch the Electron application with either the development or production build.
 * @param arg Parameter object
 * @param arg.buildType Determines if the development or the production build is used for testing
 * @param arg.colorScheme Emulates 'prefers-colors-scheme' media feature, supported values are 'light', 'dark', 'no-preference'. See page.emulateMedia([options]) for more details. Passing null resets emulation to system defaults. Defaults to 'light'.
 * @returns A promise that will resolve to an Electron application representation
 */
export const launchElectron = ({
    buildType,
    colorScheme,
}: {
    buildType?: 'development' | 'production'
    colorScheme?: null | 'light' | 'dark' | 'no-preference'
} = {}) => {
    const arg =
        buildType === 'production' ? './build/production/main/main.js' : '.'

    return electron.launch({
        args: [arg],
        colorScheme,
    })
}
