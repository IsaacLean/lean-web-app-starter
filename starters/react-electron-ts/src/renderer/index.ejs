<!doctype html>
<html>
    <head>
        <meta charset="utf-8" />
        <title><%= htmlWebpackPlugin.options.title %></title>
        <meta name="viewport" content="width=device-width,initial-scale=1" />
    </head>
    <body>
        <div id="root"></div>
        We are using Node.js <span id="node-version"></span>, Chromium
        <span id="chrome-version"></span>, and Electron
        <span id="electron-version"></span>.
    </body>
</html>
