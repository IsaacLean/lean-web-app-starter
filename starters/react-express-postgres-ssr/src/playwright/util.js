/**
 * Load the frontend application with either the development or end-to-end host.
 * By default the PLAYWRIGHT_HOST environment variable will be used. If the E2E environment
 * variable is true, then the PLAYWRIGHT_HOST_E2E environment variable will be used instead.
 * You can override these environment variables through some of the available parameters.
 * @param page Playwright Page instance
 * @param arg Parameter object
 * @param arg.gotoOptions Options to pass to Page.goto method
 * @param arg.serverType Determines if the development or end-to-end host is used for testing. Overrides E2E environment variable.
 * @param arg.url Manually set the URL. Overrides all behaviors caused by environment variables.
 * @returns A promise that will resolve to the main resource response
 */
export const loadApp = (page, { gotoOptions, serverType, url } = {}) => {
    let u = process.env.E2E
        ? process.env.PLAYWRIGHT_HOST_E2E
        : process.env.PLAYWRIGHT_HOST
    if (url) {
        u = url
    } else if (process.env.PLAYWRIGHT_HOST_E2E && serverType === 'e2e') {
        u = process.env.PLAYWRIGHT_HOST_E2E
    }

    if (!u) {
        throw new Error('Host was not set')
    }

    return page.goto(u, gotoOptions)
}
