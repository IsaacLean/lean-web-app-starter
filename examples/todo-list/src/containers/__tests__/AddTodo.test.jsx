import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import thunk from 'redux-thunk'

import todos from '../../reducers'
import { setupStore } from '../../util/store'
import AddTodo from '../AddTodo'

describe('AddTodo', () => {
    it('renders properly', () => {
        const container = renderer.create(
            <Provider store={setupStore(todos, null, [thunk])}>
                <AddTodo />
            </Provider>,
        )
        expect(container).toMatchSnapshot()
    })
})
