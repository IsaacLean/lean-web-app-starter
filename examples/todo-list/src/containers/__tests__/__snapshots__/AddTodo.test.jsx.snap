// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo renders properly 1`] = `
[
  <input />,
  <button
    onClick={[Function]}
  >
    Add Todo
  </button>,
]
`;
